### ISSUE TITLE
Training of the second stage

### ISSUE THREAD

**Issue by JZArray** [KwaiVGI/LivePortrait#350]
Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?

**Comment by Apokar** [comment]
Hi,May i ask how did you train this model. Did you write the training code and collect train-data yourself?

**Comment by JZArray** [comment]
> Hi,May i ask how did you train this model. Did you write the training code and collect train-data yourself?

Yes, we have implemented training codes and collected datasets by ourselves.

**Comment by henbucuoshanghai** [comment]
implemented training codes，great

**Comment by Vincent-luo** [comment]
@JZArray Hi, I want to ask do you know how to train on images? the paper says they train the model on both images and videos, but image just has one frame, so we just set source image and target image the same? Then will the model learn something useful because it can just copy and paste the image?

**Comment by JZArray** [comment]
> @JZArray Hi, I want to ask do you know how to train on images? the paper says they train the model on both images and videos, but image just has one frame, so we just set source image and target image the same? Then will the model learn something useful because it can just copy and paste the image?

Same ID but different two frames as source and target respectively

**Comment by JZArray** [comment]
> Hi @JZArray, can you share the training code and sample dataset that needs for training

Sry, I can't upload my codes and datasets...

**Comment by renrenzsbbb** [comment]
> > Hi,May i ask how did you train this model. Did you write the training code and collect train-data yourself?
> 
> Yes, we have implemented training codes and collected datasets by ourselves.

how can you to get sholder mask? and can you can considerable result as author pretrained model?

**Comment by lmpeng12** [comment]
> > 您好，您能否分享一下训练所需的训练代码和示例数据集
> 
> 对不起，我无法上传我的代码和数据集...

Hi,May i ask how did you train this model. Did you write the training code ？

**Comment by lmpeng12** [comment]
> > Hi @JZArray, can you share the training code and sample dataset that needs for training
> 
> Sry, I can't upload my codes and datasets...

Hi,May i ask how did you train this model. Did you write the training code ？

**Comment by xLuge** [comment]
> Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?

hi, How did you create the dataset？use  ”crop_driving_video()“ for a video or "crop_source_image" for each frame。

**Comment by ZardZen** [comment]
> > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> 
> hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。

Hi, i have the same question too. Have you saved the problem?

**Comment by luoxyhappy** [comment]
> > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > 
> > 
> > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> 
> Hi, i have the same question too. Have you saved the problem?

I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.

**Comment by xLuge** [comment]
> > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > 
> > > 
> > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > 
> > 
> > Hi, i have the same question too. Have you saved the problem?
> 
> I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.

I did it this way, but the fine-tuning results were terrible.

**Comment by luoxyhappy** [comment]
> > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > 
> > > > 
> > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > 
> > > 
> > > Hi, i have the same question too. Have you saved the problem?
> > 
> > 
> > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> 
> I did it this way, but the fine-tuning results were terrible.

Do you follow this codebase? https://github.com/zhanglonghao1992/One-Shot_Free-View_Neural_Talking_Head_Synthesis

**Comment by xLuge** [comment]
> > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > 
> > > > > 
> > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > 
> > > > 
> > > > Hi, i have the same question too. Have you saved the problem?
> > > 
> > > 
> > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > 
> > 
> > I did it this way, but the fine-tuning results were terrible.
> 
> Do you follow this codebase? https://github.com/zhanglonghao1992/One-Shot_Free-View_Neural_Talking_Head_Synthesis

Yes, and I used the MEAD and VFHQ datasets.

**Comment by JZArray** [comment]
> > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > 
> > > 
> > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > 
> > 
> > Hi, i have the same question too. Have you saved the problem?
> 
> I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.

Nope

**Comment by JZArray** [comment]
> > > Hi [@JZArray](https://github.com/JZArray), can you share the training code and sample dataset that needs for training
> > 
> > 
> > Sry, I can't upload my codes and datasets...
> 
> Hi,May i ask how did you train this model. Did you write the training code ？

Yes, we write our own training codes

**Comment by luoxyhappy** [comment]
> > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > 
> > > > 
> > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > 
> > > 
> > > Hi, i have the same question too. Have you saved the problem?
> > 
> > 
> > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> 
> Nope

Do you have any suggestions on crop method? Thank you very much.

**Comment by JZArray** [comment]
> > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > 
> > > > > 
> > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > 
> > > > 
> > > > Hi, i have the same question too. Have you saved the problem?
> > > 
> > > 
> > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > 
> > 
> > Nope
> 
> Do you have any suggestions on crop method? Thank you very much.

You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames

**Comment by luoxyhappy** [comment]
> > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > 
> > > > > > 
> > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > 
> > > > > 
> > > > > Hi, i have the same question too. Have you saved the problem?
> > > > 
> > > > 
> > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > 
> > > 
> > > I did it this way, but the fine-tuning results were terrible.
> > 
> > 
> > Do you follow this codebase? https://github.com/zhanglonghao1992/One-Shot_Free-View_Neural_Talking_Head_Synthesis
> 
> Yes, and I used the MEAD and VFHQ datasets.

Could you please show some samples of your model? Thanks

**Comment by JZArray** [comment]
> > > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > > 
> > > > > > > 
> > > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > > 
> > > > > > 
> > > > > > Hi, i have the same question too. Have you saved the problem?
> > > > > 
> > > > > 
> > > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > > 
> > > > 
> > > > I did it this way, but the fine-tuning results were terrible.
> > > 
> > > 
> > > Do you follow this codebase? https://github.com/zhanglonghao1992/One-Shot_Free-View_Neural_Talking_Head_Synthesis
> > 
> > 
> > Yes, and I used the MEAD and VFHQ datasets.
> 
> Could you please show some samples of your model? Thanks

Sry, I can't upload them

**Comment by luoxyhappy** [comment]
> > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > 
> > > > > > 
> > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > 
> > > > > 
> > > > > Hi, i have the same question too. Have you saved the problem?
> > > > 
> > > > 
> > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > 
> > > 
> > > Nope
> > 
> > 
> > Do you have any suggestions on crop method? Thank you very much.
> 
> You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames

If the bbox is fixed based on the first frame, is it possible that the head could move outside the bbox in subsequent frames？

**Comment by xLuge** [comment]
> > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > 
> > > > > > 
> > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > 
> > > > > 
> > > > > Hi, i have the same question too. Have you saved the problem?
> > > > 
> > > > 
> > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > 
> > > 
> > > Nope
> > 
> > 
> > Do you have any suggestions on crop method? Thank you very much.
> 
> You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames

hi, did you add Region Loss and Wing Loss, and which ten points did you choose?

**Comment by JZArray** [comment]
> > > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > > 
> > > > > > > 
> > > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > > 
> > > > > > 
> > > > > > Hi, i have the same question too. Have you saved the problem?
> > > > > 
> > > > > 
> > > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > > 
> > > > 
> > > > Nope
> > > 
> > > 
> > > Do you have any suggestions on crop method? Thank you very much.
> > 
> > 
> > You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames
> 
> If the bbox is fixed based on the first frame, is it possible that the head could move outside the bbox in subsequent frames？

yes, it is possible, so you need to ensure there are no large head motions

**Comment by JZArray** [comment]
> > > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > > 
> > > > > > > 
> > > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > > 
> > > > > > 
> > > > > > Hi, i have the same question too. Have you saved the problem?
> > > > > 
> > > > > 
> > > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > > 
> > > > 
> > > > Nope
> > > 
> > > 
> > > Do you have any suggestions on crop method? Thank you very much.
> > 
> > 
> > You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames
> 
> hi, did you add Region Loss and Wing Loss, and which ten points did you choose?

yes, there is a issue asking the same question, and author explained it, you can have a check

**Comment by xLuge** [comment]
> > > > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > > > 
> > > > > > > > 
> > > > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > > > 
> > > > > > > 
> > > > > > > Hi, i have the same question too. Have you saved the problem?
> > > > > > 
> > > > > > 
> > > > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > > > 
> > > > > 
> > > > > Nope
> > > > 
> > > > 
> > > > Do you have any suggestions on crop method? Thank you very much.
> > > 
> > > 
> > > You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames
> > 
> > 
> > hi, did you add Region Loss and Wing Loss, and which ten points did you choose?
> 
> yes, there is a issue asking the same question, and author explained it, you can have a check

Thank you. Can you reveal how many ID videos were used?

**Comment by luoxyhappy** [comment]
> > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > 
> > > > > > 
> > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > 
> > > > > 
> > > > > Hi, i have the same question too. Have you saved the problem?
> > > > 
> > > > 
> > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > 
> > > 
> > > I did it this way, but the fine-tuning results were terrible.
> > 
> > 
> > Do you follow this codebase? https://github.com/zhanglonghao1992/One-Shot_Free-View_Neural_Talking_Head_Synthesis
> 
> Yes, and I used the MEAD and VFHQ datasets.

I'm also trying to fine-tune this model. Could you please provide your email address for further communication?

> > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > 
> > > > > > 
> > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > 
> > > > > 
> > > > > Hi, i have the same question too. Have you saved the problem?
> > > > 
> > > > 
> > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > 
> > > 
> > > Nope
> > 
> > 
> > Do you have any suggestions on crop method? Thank you very much.
> 
> You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames

Sorry, I don't quite understand the purpose of doing this kind of coping. Are you trying to align the first frame with the crop_source_image? In that case, during training, can we only use the first frame of each video as the source?

**Comment by luoxyhappy** [comment]
> > > > > > > Hello, a quick question, I am trying to train this model from scratch, and it seems the first-stage training is ok, and I have gotten a promising result. While training the second stage, results are always noised, so I want to ask how to train the second one? Should a pretrained model from the first stage be frozen, and only the stitching module is going to be trained? Whether only using stitching loss to supervise stitching module is enough, or losses used in the first stage should be also used?
> > > > > > 
> > > > > > 
> > > > > > hi, How did you create the dataset？use ”crop_driving_video()“ for a video or "crop_source_image" for each frame。
> > > > > 
> > > > > 
> > > > > Hi, i have the same question too. Have you saved the problem?
> > > > 
> > > > 
> > > > I have the same problem. I am not sure but I think they use ”crop_driving_video()“ for a video.
> > > 
> > > 
> > > Nope
> > 
> > 
> > Do you have any suggestions on crop method? Thank you very much.
> 
> You can use any detection method to first ensure a person's position (face position), then fix the bbox to do crop for all frames

Sorry, I don't quite understand the purpose of doing this kind of coping. Are you trying to align the first frame with the crop_source_image? In that case, during training, can we only use the first frame of each video as the source?
